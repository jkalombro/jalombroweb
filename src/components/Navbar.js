import React, { useCallback, useEffect } from 'react';
import { Link, Outlet, useLocation } from 'react-router-dom';
import { UseAppContext } from '../context/AppContext';
import myLogo from '../images/mylogowhite.png';

const Navbar = () => {
  const { activemenu } = UseAppContext();

  return (
    <header className="App-header">
        <div className="header-box-1">
            <img src={myLogo} alt="My Logo" />
        </div>

        {/* Main Menu */}
        <div className="header-box-2">
            <Link to="/jalombroweb/home">
                <button className={activemenu === '/jalombroweb/home' ? 'header-btn btn-selected' : 'header-btn'}>HOME</button>
            </Link>
            <Link to="/jalombroweb/skills">
                <button className={activemenu === '/jalombroweb/skills' ? 'header-btn btn-selected' : 'header-btn'}>SKILLS</button>
            </Link>
            <Link to="/jalombroweb/contact">
                <button className={activemenu === '/jalombroweb/contact' ? 'header-btn btn-selected' : 'header-btn'}>CONTACT</button>
            </Link>
        </div>
    </header>
  )
}

export const NavbarRouteWrapper = () => 
{
    const { activemenu, setActivemenu } = UseAppContext();
    const location = useLocation();

    const evaluateActiveMenu = useCallback(() => {
        if(activemenu !== location.pathname) {
            setActivemenu(location.pathname);
        }
    }, [activemenu, location, setActivemenu]);

    useEffect(() => {
        evaluateActiveMenu();
    }, [activemenu, evaluateActiveMenu]);

    return (
        <Outlet />
    );
}

export default Navbar;
