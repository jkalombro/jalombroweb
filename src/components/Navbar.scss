@import '../App.scss';

.App-header {
  position: sticky;
  position: -webkit-sticky;
  top: 0;
  background-color: #282c34;
  min-height: 10vh;
  display: flex;
  align-items: center;
  font-size: calc(10px + 2vmin);
  color: white;
  z-index: 1041;
}

.header-box-1 {
  flex: 1;
  padding-left: 50px;

  img {
    height: 6vh;
    opacity: 0.9;
  }
}

.header-box-2 {
  flex: 1;
  position: absolute;
  right: 50px;
}

a {
  .header-btn {
    font-family: "Lato", sans-serif;
    background: transparent;
    border: 0px;
    cursor: pointer;
    font-size: 20px;
    color: white;
    margin: 2vmin;
    border-bottom: 2px solid transparent;
  }

  .header-btn:hover,
  .header-btn:active {
    color: skyblue;
    border-bottom: 2px solid skyblue;
  }

  .header-btn:focus,
  .btn-selected {
    color: skyblue;
  }

  button:focus {
    outline: 0;
  }
}

@include mobile {
  .header-box-1 {
    padding-left: 30px;

    img {
      height: 3vh;
    }
  }

  .header-box-2 {
    right: 30px;

    button {
      font-size: 15px;
    }
  }
}
