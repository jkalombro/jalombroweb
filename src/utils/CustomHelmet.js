import React, { useEffect, useState } from "react";
import { UseAppContext } from "../context/AppContext";
import { Helmet } from "react-helmet";
const meta = require("./metatags.json");

const CustomHelmet = () => {
    const { activemenu } = UseAppContext();
    const [title, setTitle] = useState(meta.title);

    useEffect(() => {
        switch (activemenu) {
            case "/":
                setTitle(meta.title);
                break;
            case "/aboutme":
                setTitle(`About Jk`);
                break;
            case "/contact":
                setTitle(`Contact Jk`);
                break;
            default: {
            }
        }
    }, [activemenu]);

    return (
        <Helmet>
            {/* Primary Meta Tags */}
            <title>{title}</title>
            <meta name="title" content={title} />
            <meta name="description" content={meta.description} />

            {/* Open Graph / Facebook  */}
            <meta property="og:type" content={meta.type} />
            <meta property="og:url" content={meta.site_url} />
            <meta property="og:title" content={title} />
            <meta property="og:description" content={meta.description} />
            <meta property="og:image" content={meta.img_url} />

            {/* Twitter */}
            <meta property="twitter:card" content={meta.twitter_card} />
            <meta property="twitter:url" content={meta.site_url} />
            <meta property="twitter:title" content={title} />
            <meta property="twitter:description" content={meta.description} />
            <meta property="twitter:image" content={meta.img_url} />
        </Helmet>
    );
};

export default CustomHelmet;
