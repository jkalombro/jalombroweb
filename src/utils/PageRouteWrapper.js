import React, { useCallback, useEffect, Suspense } from 'react';
import { Outlet, useLocation } from 'react-router-dom';
import { UseAppContext } from '../context/AppContext';
import Loading from '../components/Loading';

// This utility component helps the page components perform all the common things they need to do when rendered
const PageRouteWrapper = () => 
{
    const { activemenu, setActivemenu } = UseAppContext();
    const location = useLocation();

    const evaluateActiveMenu = useCallback(() => {
        if(activemenu !== location.pathname) {
            setActivemenu(location.pathname);
        }
    }, [activemenu, location, setActivemenu]);

    useEffect(() => {
        evaluateActiveMenu();
    }, [activemenu, evaluateActiveMenu]);

    return (
        <Suspense fallback={<Loading />}>
            <Outlet />
        </Suspense>
    );
}

export default PageRouteWrapper;