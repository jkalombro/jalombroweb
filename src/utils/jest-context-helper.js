import { render as rtlRender } from "@testing-library/react";
import { AppContextProvider } from '../context/AppContext';

//This utility component helps the testers to render components wrapped by ContextAPI provider
const render = component => rtlRender(
    <AppContextProvider>
      {component}
    </AppContextProvider>
  )

export default render;