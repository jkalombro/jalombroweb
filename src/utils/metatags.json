{
    "title": "Jk Alombro",
    "description": "My simple portfolio website built using ReactJS.",
    "img_url": "https://i.imgur.com/vMLsV1Y.png",
    "site_url": "https://jkalombro.herokuapp.com",
    "type": "website",
    "twitter_card": "summary_large_image"
}
