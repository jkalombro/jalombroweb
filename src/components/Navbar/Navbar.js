import React from "react";
import { Link } from "react-router-dom";
import { UseAppContext } from "../../context/AppContext";
import myLogo from "../../images/mylogowhite.png";
import "./Navbar.scss";

const Navbar = () => {
    const { activemenu } = UseAppContext();

    return (
        <header className="App-header" data-testid="navbar">
            <div className="header-box-1">
                <img src={myLogo} alt="My Logo" />
            </div>

            {/* Main Menu */}
            <div className="header-box-2">
                <Link to="/">
                    <button className={activemenu === "/" ? "header-btn btn-selected" : "header-btn"}>HOME</button>
                </Link>
                <Link to="/aboutme">
                    <button className={activemenu === "/aboutme" ? "header-btn btn-selected" : "header-btn"}>ABOUT ME</button>
                </Link>
                <Link to="/contact">
                    <button className={activemenu === "/contact" ? "header-btn btn-selected" : "header-btn"}>CONTACT</button>
                </Link>
            </div>
        </header>
    );
};

export default Navbar;
