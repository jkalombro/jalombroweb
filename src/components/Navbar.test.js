import React from 'react';
import '@testing-library/jest-dom';
import { renderRouter } from '../utils/jest-context-helper';
import Navbar from './Navbar';

describe("Navbar Component", () => {
  it('renders without crashing', async () => {
    const { getByTestId } = renderRouter(<Navbar />);
    const element = getByTestId("navbar");
    expect(element).toBeTruthy();
  });

  it('matches with snapshot', async () => {
    const { getByTestId } = renderRouter(<Navbar />);
    const element = getByTestId("navbar");
    expect(element).toMatchSnapshot();
  });

  it('has logo', async () => {
    const { getByAltText } = renderRouter(<Navbar />);
    const element = getByAltText(/My Logo/i);
    expect(element).toBeInTheDocument();
  });

  it('has all menu', async () => {
    const { getByText } = renderRouter(<Navbar />);
    const home = getByText(/home/i);
    const aboutme = getByText(/about me/i);
    const contact = getByText(/contact/i);
    expect(home).toBeInTheDocument();
    expect(aboutme).toBeInTheDocument();
    expect(contact).toBeInTheDocument();
  });
  
});