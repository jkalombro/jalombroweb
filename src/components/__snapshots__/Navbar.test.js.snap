// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component matches with snapshot 1`] = `
<header
  class="App-header"
  data-testid="navbar"
>
  <div
    class="header-box-1"
  >
    <img
      alt="My Logo"
      src="mylogowhite.png"
    />
  </div>
  <div
    class="header-box-2"
  >
    <a
      href="/jalombroweb"
    >
      <button
        class="header-btn btn-selected"
      >
        HOME
      </button>
    </a>
    <a
      href="/jalombroweb/aboutme"
    >
      <button
        class="header-btn"
      >
        ABOUT ME
      </button>
    </a>
    <a
      href="/jalombroweb/contact"
    >
      <button
        class="header-btn"
      >
        CONTACT
      </button>
    </a>
  </div>
</header>
`;
